### PR TITLE
[Ltac2] “constr” arguments to tactic notations may have interpretation scopes

### PR DESCRIPTION
**Pull request by vbgl**
This is a possible solution to the feature requested in #4381.

This defines a “delimited” scope which takes as argument an identifier: the delimiter of the scope in which the tactic argument (a constr) is to be interpreted.

@JasonGross (original author of the feature request): would this address your needs?

@ppedrot (ltac2 expert): is this the kind of implementation you had in mind? Suggestions, comments welcome.

TODO:
- [x] Added / updated test-suite
- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

**Comment by JasonGross** [comment]
It is definitely a step in the right direction and satisfies the bulk of my request.  There are two pieces missing, though:
1. There is no support for interpreting in the scope bound to the type of the goal (though perhaps this is not a reasonable request)
2. There is no support for multiple scopes in the scope stack (`%string%list`), though I expect I'd only rarely want this.

**Comment by ppedrot** [comment]
@vbgl this is indeed what I had in mind. I think I would have reused the `constr` scope name, but I guess it's fine to use another name for it.

**Comment by vbgl** [comment]
Thanks for the feed-back. I’ll reuse the `constr` name and add support for multiple notation scopes.

Regarding the “goal” pseudo-scope, this seems to be a different issue. In particular, at parsing time, there is no goal nor environment to explore.

**Comment by maximedenes** [comment]
@ppedrot can you assign yourself? Thanks!

File: user-contrib/Ltac2/tac2quote.mli [hunk 1]
```diff
@@ -32,7 +32,7 @@ val of_variable : Id.t CAst.t -> raw_tacexpr
 
 val of_ident : Id.t CAst.t -> raw_tacexpr
 
-val of_constr : Constrexpr.constr_expr -> raw_tacexpr
+val of_constr : Id.t list -> Constrexpr.constr_expr -> raw_tacexpr
```
Comment (ppedrot) [inline user-contrib/Ltac2/tac2quote.mli]: Nitpicking, but this could be an optional argument called `scopes` so that it reduces the unrelated changes.

Comment (vbgl) [reply user-contrib/Ltac2/tac2quote.mli]: Sounds like a good idea. I’m unsure about the “scopes” name: it is unclear whether it refers to ltac2 scopes or Coq notation scopes.

What about “delimiters”?

Comment (ppedrot) [reply user-contrib/Ltac2/tac2quote.mli]: Fine.

Comment (vbgl) [reply user-contrib/Ltac2/tac2quote.mli]: Done.

**Review by Zimmi48 — APPROVED** [review]

**Comment by ppedrot** [comment]
CI failure spurious, will merge.

**Review by ppedrot — APPROVED** [review]

**Comment by RalfJung** [comment]
Cc @robbertkrebbers we could also really use this for Iris.
